tapalcatl_server/server.go

```diff
@@ -333,15 +331,11 @@ func main() {
 	// metrics writer configuration
 	var mw metricsWriter
 	if metricsStatsdAddr != "" {
-		if metricsInterval <= 0 {
-			logger.Fatalf("ERROR: metrics interval should be greater than 0")
-		}
 		udpAddr, err := net.ResolveUDPAddr("udp4", metricsStatsdAddr)
 		if err != nil {
 			logger.Fatalf("ERROR: Invalid metricsstatsdaddr %s: %s", metricsStatsdAddr, err)
 		}
-		metricsIntervalDuration := time.Second * time.Duration(metricsInterval)
-		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, metricsIntervalDuration)
+		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, logger)
```

Comment:
If the main thread exits, then the process has terminated and we're finished anyway right? Were you thinking of putting the close after the server is done? Or somewhere else? Here's an example of exiting without waiting for a goroutine: https://play.golang.org/p/b3lfyqTv2v

There isn't any other explicit cleanup that we need to do besides the garbage collection, so I don't think it matters either way.